src/components/ChatContent/Messages/IrcMessage.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <message
```

Comment:
I don't think I necessarily agree. At least how it is now, IrcMessage is focused around a Message object, meaning that it won't work well with `ConnectionMessage` (and wouldn't be ideal for how I use it in `OnlineTab`). I think that there is value in separating "A thing that is within the message list" from "An IRC message".
Perhaps I should rename it if we end up keeping it.

Reply:
Well it definitely doesn't make sense for an item in the users tab to be a "message" - dunno whether that's just a "needs a different name" or if they're really two discrete things that may look similar, but should really be controlled separately.

I dunno what I think about having an entire component just to add a class, even if it'll be used in multiple places. Like "there's this message component, what responsibility does that even have that this component doesn't, and now I have to look someplace else to understand what this component does even though that other component is pretty much trivial". Honestly I'm generally wary about faking inheritance by using composition (ie, an IRCMessage being a Message component with some stuff inside it - similar thoughts with Tab). Could very well be that I'm just not used to it, but to be it seems to add a lot of this kind of complexity.

Ultimately I don't have any concrete suggestions. You might be right that this is the way to do it. Just food for thought in case you have other ideas for how to architect this that'd avoid those issues I guess.

Reply:
I think it just needs a different name, due to the fact that we chose* to have the users and the messages tabs look and function in the same way.
*Did we ever make that a conscious decision? Is there any reason to change that before implementing a discord-style system?

Reply:
Do you think that chat-item would be a good name for this?

Reply:
> Did we ever make that a conscious decision? Is there any reason to change that before implementing a discord-style system?

Honestly, I want this to go away ASAP. :P I shared the designs Sharif came up with at one point:
![image](https://user-images.githubusercontent.com/18635705/62265641-e364c380-b3f2-11e9-960e-d94e5e2a81e2.png)
![image](https://user-images.githubusercontent.com/18635705/62265647-e790e100-b3f2-11e9-9ac8-5ca52243b4ef.png)

chat-item seems like a fine name if you don't want to go after reimplementing the new design. Alternatively, just make a message and a user list item two completely separate things.

Reply:
I don't even use the color from that class on the username... Dropped